server/include/bge/System.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "Entity.h"
+#include "ComponentManager.h"
+#include "Component.h"
+#include <set>
+#include <bitset>
+
+namespace bge {
+    class System {
+        public:
+            void init();
+            void update(int dt);
```

Comment:
What does this function do? What does dt represent? Right now the implementation is empty, so I'm just curious how it fits into things.

Reply:
Basically for every iteration of the game loop the update function is called on all systems in the game world. I believe dt is supposed to represent the tick time, which might come in handy when calculating movement. To be fair I copied this part directly from the tutorial, so we might not need the dt parameter.

Reply:
^yep, we don't need to worry about delta_time here because the server tick rate is constant.

Reply:
There will be a variety of systems: CollisionSystem, FrozenSystem, etc. 

Each system class (e.g. FrozenSystem)'s update() will grab all Entites with the relevant component (e.g. FrozenComponent) and modify them.